### PR TITLE
fix(datepicker): prevent `matInput` from clobbering date value

### LINKED ISSUE #7717: (Datepicker) Initial Value giving full datetime
#### Bug, feature request, or proposal:
Bug/Support

#### What is the expected behavior?
When I send in a new moment or date object into the datepicker it should format like it does when you click a new date.

#### What is the current behavior?
The initial format is the full datetime string instead of MM/DD/YYYY. It might be I am not sending in it in right or something but there is no documentation I can find that says to do it different.

#### What are the steps to reproduce?
https://stackblitz.com/edit/angular-material2-issue-y6hdoi

I have tried new Date('12/06/2017') as well and still get the full datetime string.

#### What is the use-case or motivation for changing an existing behavior?
To have the datepicker set a value on init that is from data but formatted the way the datepicker selects dates.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
5.0.0-rc.1

### PR DESCRIPTION
**Pull request by mmalerba**
fixes #7717

### CODE REVIEW

File: src/lib/input/input-value-accessor.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {InjectionToken} from '@angular/core';
+
+
+export const MAT_INPUT_VALUE_ACCESSOR =
```
Comment (crisbeto) [inline src/lib/input/input-value-accessor.ts]: Add a doc describing what this is for?

File: src/lib/input/input-value-accessor.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    new InjectionToken<{value: any}>('MAT_INPIUT_VALUE_ACCESSOR');
```
Comment (rafaelss95) [inline src/lib/input/input-value-accessor.ts]: typo?

**Comment by mmalerba** [comment]
Done

**Comment by kiancross** [comment]
Any ETA for when this will be merged?

**Pull request merged**